node/src/services/builder.rs

```diff
@@ -424,11 +440,16 @@ where
         max_storage_capacity: Option<StorageDataUnit>,
         jump_capacity: Option<StorageDataUnit>,
         extrinsic_retry_timeout: u64,
+        msp_charging_freq: Option<u32>,
     ) {
         self.setup_storage_layer(storage_path);
         if max_storage_capacity.is_none() {
             panic!("Max storage capacity not set");
         }
+        if let Some(notify_period) = msp_charging_freq {
+            self.with_notify_period(notify_period.clone());
+        }
+
```

Comment:
I am not super happy for having to do this with all different `setup` function. It is a bit redundant with the builder strategy.

Reply:
Agreed. In fact all the functions that are not storage-layer-or-provider-type-specific (i.e. that they don't need to have a special implementation for a provider type and storage layer), I'd leave them out of this `setup`.

In other words, to make proper use of the builder pattern I would:
1. Remove the use of `with_notify_period` from here, and call it in `service.rs` (in `init_sh_builder()`), in case the CLI parameter was passed.
2. Do the same with `with_max_storage_capacity`, `with_jump_capacity`, `with_retry_timeout`, as they seem to be called in the same way, regarless of what the provider type and storage layer is. Well almost, in the User is different. But anyway, these tiny initialisation differences are exactly what the builder pattern is meant to be good for.

Reply:
Anyway, I'd like to get @snowmead involved since he was heavily involved in this code.